rest/config.go

```diff
@@ -1751,6 +1766,21 @@ func (sc *ServerContext) applyConfigs(ctx context.Context, dbNameConfigs map[str
 
 // _applyConfig loads the given database, failFast=true will not attempt to retry connecting/loading
 func (sc *ServerContext) _applyConfig(nonContextStruct base.NonCancellableContext, cnf DatabaseConfig, failFast, isInitialStartup bool) (applied bool, err error) {
+	ctx := nonContextStruct.Ctx
+
+	registry, err := sc.BootstrapContext.getGatewayRegistry(ctx, *cnf.Bucket)
```

Comment:
The idea behind the ConfigManager interface was to try to abstract/hide the underlying registry implementation from ServerContext.  I'm realizing now I already broke that contract in handleGetClusterInfo, but I think we'd benefit from keeping the _applyConfig code as simple as possible, and turn the two blocks of new code here into:
```
  ConfigManager.VerifyVersion(ctx, bucketName, currentVersion) error
  ConfigManager.SetVersion(ctx. bucketName, currentVersion)
```  
What do you think?